WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift

```diff
@@ -631,4 +634,25 @@ extension WPAnalytics {
         WPAnalytics.track(stat, withProperties: props)
     }
 
+    /// This will call each registered tracker and fire the given event.
+    /// - Parameters:
+    ///   - eventName: a `String` that represents the event name
+    ///   - properties: a `Hash` that represents the properties
+    ///   - blog: a `Blog` asssociated with the event
+    static func trackBlockEditorEvent(_ eventName: String, properties: [AnyHashable: Any], blog: Blog) {
+        var props = properties
+        props[WPAppAnalyticsKeyBlogID] = blog.dotComID
+        props[WPAppAnalyticsKeySiteType] = blog.isWPForTeams() ? WPAppAnalyticsValueSiteTypeP2 : WPAppAnalyticsValueSiteTypeBlog
+
+        var event: WPAnalyticsEvent?
+        switch eventName {
+        case "editor_block_inserted": event = .gutenbergEditorBlockInserted
+        default: event = nil
+        }
```

Comment:
This switch statement manages the event name string to `WPAnalyticEvent` conversion. It is currently placed here in an attempt to keep any analytic type-related work within this one file.